### PR TITLE
[DO NOT MERGE]: Example of multi column "mega menu"

### PR DESCRIPTION
**Pull request by wodenx**
This demonstrates how to implement a multi-column menu.
It is very much a POC and requires significant refactoring before inclusion -- but should serve well enough for a pilot site.  Eventually, after refactoring, can be included in bodiless as part of the default menu.

### CODE REVIEW

File: examples/test-site/src/components/Menus/ColumnSubList.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+// The sublist that will go in each column.
+const asColumnSubList = flow(
+  pickProps(sublistPropsToKeep),
+  asEditableList,
+  withEditableTitle(asEditable),
+);
+
+// TODO: Make this generic: allow alternate component to render the item with sublist.
+// TODO: Refactor to use rc-menu MenuGroup rather than embedded sublist.
+// This duplicates a lot of the code in withSublist.  We need to improve that code
```
Comment (beliayeu) [inline examples/test-site/src/components/Menus/ColumnSubList.tsx]: at the first glance, once we refactor to use rc-menu MenuGroup, then we will have to switch to withSubmenuToggle.

File: examples/test-site/src/components/Menus/ColumnSubList.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    const ItemWithSublist: FC<ListProps> = ({
+      children, unwrap, nodeKey, ...rest
+    }) => (
+      <Item {...rest}>
+        {children}
+        <Sublist unwrap={unwrap} nodeKey="sublist" />
+      </Item>
+    );
+    // TODO: Why is this type-cast necessary?
+    return withToggleTo(Item)(addClasses('inline-block')(ItemWithSublist as ComponentType));
```
Comment (beliayeu) [inline examples/test-site/src/components/Menus/ColumnSubList.tsx]: to my mind we should not add classes here

**Comment by beliayeu** [comment]
@hvanyo, @wodenx, do we have plans to update burger menu as part of this ticket? currently, burger menu display the first 2 levels (the 3rd level is not displayed)

**Comment by hvanyo** [comment]
Attached this code to our Mega Menu component so this is not  lost and can be used when we discuss mega menu.  Closing as won't continue this POC.

**Pull request closed without merging**